src/nvim/ex_docmd.c

```diff
@@ -2204,7 +2205,7 @@ find_ucmd (
             eap->cmdidx = CMD_USER;
           else
             eap->cmdidx = CMD_USER_BUF;
-          eap->argt = (long)uc->uc_argt;
+          eap->argt = uc->uc_argt;
```

Comment:
That last statement is not exactly correct. Let me rephrase:
For 64bit, uint32_t is always safely convertible to long, so everything's correct, even without cast.
For 32bit, warning would be generated, for assigning unsigned to signed of the same size.
Now, we know unsigned value never uses the last bit, so a simple cast to long would suffice.
But that would be fragile, as it would break if new values using the last bit were introduced.
So, proper solution would be to assert assigned value is not bigger than LONG_MAX.
But that would generate and "always true" warning in 64 bits.
So, definitive thing should look like:
```
assert((uintmax_t)uc->uc_argt <= LONG_MAX);
eap->argt = (long)uc->uc_argt
```

I will correct to that if you agree.